src/vaadin-grid-styling-mixin.html

```diff
@@ -0,0 +1,73 @@
+<!--
+@license
+Copyright (c) 2018 Vaadin Ltd.
+This program is available under Apache License Version 2.0, available at https://vaadin.com/license/
+-->
+<script>
+  window.Vaadin = window.Vaadin || {};
+  window.Vaadin.Grid = window.Vaadin.Grid || {};
+
+  /**
+   * @polymerMixin
+   */
+  Vaadin.Grid.StylingMixin = superClass => class StylingMixin extends superClass {
+
+    static get properties() {
+      return {
+        /**
+         * A function that allows generating CSS class names for grid cells
+         * based on their row and column. The return value should be the generated
+         * class name as a string, or multiple class names separated by whitespace
+         * characters.
+         *
+         * Receives two arguments:
+         * - `rowData` The object with the properties related with
+         *   the rendered item, contains:
+         *   - `rowData.index` The index of the item.
+         *   - `rowData.item` The item.
+         *   - `rowData.expanded` Sublevel toggle state.
+         *   - `rowData.level` Level of the tree represented with a horizontal offset of the toggle button.
+         *   - `rowData.selected` Selected state.
+         * - `column` The `<vaadin-grid-column>` element.
+         */
+        cellClassNameGenerator: Function
+      };
+    }
+
+    static get observers() {
+      return [
+        '__cellClassNameGeneratorChanged(cellClassNameGenerator)'
+      ];
+    }
+
+    __cellClassNameGeneratorChanged(cellClassGenerator) {
+      this.generateClassNames();
```

Comment:
I would prefer not to do this, because currently you can clear the generated styles by setting `cellClassNameGenerator` to a falsy value (I think there might be use-cases for this).

After the proposed change, the following code would leave the 'foo' classes to the cells:
```javascript
grid.cellClassNameGenerator = (rowData, column) => 'foo';
grid.cellClassNameGenerator = undefined;
```

But maybe I should skip the generating if both the old value and the new value are falsy? In this case it doesn't make any difference for the behavior.

Reply:
You can use single property observer, e.g. instead of `static get observers()` make it looks like this:
```js
cellClassNameGenerator: {
  type: Function,
  observer: '__cellClassNameGeneratorChanged'
}
```
there you could check for both new and old value, to make sure whether property has been reset or not.

Reply:
Done

Reply:
Note the syntax in my example above, `observer` should be reference to the class method, not the function itself (how it is in your code now). See this example: https://github.com/vaadin/vaadin-grid/blob/bc060564eb2cb09d78ea538dc8436678e4f74930/src/vaadin-grid-data-provider-mixin.html#L86

Reply:
Can you elaborate why it needs to be done in that way?

I noted the syntax but IMO it's cleaner like this, since the function is so simple and it doesn't need to add a new private function to grid (it's already painful to browse through the grid API).

I got this from the Polymer docs: https://www.polymer-project.org/2.0/docs/devguide/observers#change-callbacks (second example)
so I would expect it to be okay unless we have some common convention about this.

Reply:
> first time the complex observers are invoked with the undefined passed as an argument.

Actually I tested this, and the observer is not run with `undefined` (or anything else) initially. If I don't set the property, the observer is not called at all. If I set the property, the property value is passed as the argument on the first and only time.

So now my plan is to just revert that commit. Is it okay, or did I maybe misunderstand your original idea in this thread?

Also that observer syntax that I used seems to cause problems after Polymer3-conversion so better not to use it.

Reply:
I reverted the commit for now.